library/sha256.c

```diff
@@ -389,10 +397,16 @@ static const unsigned char sha256_test_sum[6][32] =
 int mbedtls_sha256_self_test( int verbose )
 {
     int i, j, k, buflen, ret = 0;
-    unsigned char buf[1024];
+    unsigned char *buf;
```

Comment:
In general we may want to do a lot of similar changes, maybe even in the library itself. Since this can affect bare-metal builds adversely, it may be a good idea to introduce a compile-time switch that decides if we want to allocate our buffers on the stack or on the heap.

We could have 3 stages:
- every local buffer on the stack
- mixed (as it is now/default)
- every local buffer on the heap 

I am not saying that we should do this right now, it is just an idea.